src/PrestaShopBundle/Entity/ApiAccess.php

```diff
@@ -30,60 +30,73 @@
 
 use Doctrine\ORM\Mapping as ORM;
 use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\ApiAccessInterface;
-use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\AuthorizedApplicationInterface;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * @ORM\Entity(repositoryClass="PrestaShopBundle\Entity\Repository\ApiAccessRepository")
  * @ORM\Table()
- * @UniqueEntity("name")
  *
  * @experimental
  */
+#[UniqueEntity('clientId')]
+#[UniqueEntity('clientName')]
 class ApiAccess implements ApiAccessInterface
 {
     /**
-     * @var int
-     *
      * @ORM\Id
-     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned":true})
+     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned": true})
      * @ORM\GeneratedValue(strategy="AUTO")
      */
-    private $id;
+    #[Assert\Type('integer')]
+    #[Assert\Positive]
+    private int $id;
 
     /**
-     * @var string
-     *
      * @ORM\Column(name="client_id", type="string", length=255)
      */
-    private $clientId;
+    #[Assert\Type('string')]
+    #[Assert\Length(max: 255)]
+    #[Assert\NotBlank]
+    private string $clientId;
 
     /**
-     * @var string
-     *
-     * @ORM\Column(name="client_secret", type="string", length=255)
+     * @ORM\Column(name="client_name", type="string", length=255)
      */
-    private $clientSecret;
+    #[Assert\Type('string')]
+    #[Assert\Length(max: 255)]
+    #[Assert\NotBlank]
+    private string $clientName;
 
     /**
-     * @var AuthorizedApplicationInterface
+     * We make the secret nullable for the moment because it prevents the first step of the feature to be implemented.
      *
-     * @ORM\ManyToOne(targetEntity=AuthorizedApplication::class)
-     * @ORM\JoinColumn(name="id_authorized_application", referencedColumnName="id_authorized_application", nullable=false, onDelete="CASCADE")
+     * @ORM\Column(name="client_secret", type="string", length=255, nullable=true)
      */
-    private $authorizedApplication;
+    #[Assert\Type(['string', 'null'])]
+    #[Assert\Length(max: 255)]
+    private string $clientSecret;
```

Comment:
```suggestion
    private ?string $clientSecret;
```
If it's nullable for now, so should the type